temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -446,13 +456,13 @@ public long getLastStartedEventId() {
             attributes,
             (p, f) -> {
               callback.apply(p, f);
-              if (f != null && f.getCause() != null && f.getCause().hasCanceledFailureInfo()) {
+              if (f != null && f.getCause().hasCanceledFailureInfo()) {
```

Comment:
Are you 100% sure that it is safe to expect the cause for all possible failures?

Reply:
It's protobuf, so f.getCause() != null is always true, that check didn't make sense. But... I just added an additional `f.hasCause()` check.